zproject/email_backends.py

```diff
@@ -108,3 +109,9 @@ def send_messages(self, email_messages: Sequence[EmailMessage]) -> int:
                 email_log_url = settings.ROOT_DOMAIN_URI + "/emails"
                 logging.info("Emails sent in development are available at %s", email_log_url)
         return num_sent
+
+
+class CustomSMTPEmailBackend(EmailBackend):
```

Comment:
Yes, it's custom -- but this name doesn't tell me _what about it is custom_.  I think we can come up with a better name that describes what this does. :)

Reply:
And if we're going to use this as our EmailBackend -- can we move the connection-tracking logic from `initialize_connection` onto this class?  Having a `isinstance(connection, CustomSMTPEmailBackend)` where _we wrote the class_ is a smell that we're possibly not actually using good object orientation.

Reply:
This will our new `SMTP` email backend but not BaseEmailBackend. I'm wondering whether it would be appropriate to handle `EmailLogBackend` logic in a SMTP email backend class.

Reply:
I think you accidentally a word?

`EmailLogBackend` logic should remain separate.

Reply:
I've moved the logic to `PersistentSMTPEmailBackend` in a seperate commit, so it's easier to review and make changes for me if required, we can squash it before merging if we desire to have them in the same commit